### PR TITLE
[WFCORE-2109] Require Maven 3.3.1+ and introduce mvnw

### PR DESCRIPTION
**Pull request by ppalaga**
https://issues.jboss.org/browse/WFCORE-2109

### CODE REVIEW

**Comment by ppalaga** [comment]
These changes were presented and discussed on the F2F meeting in Brno. cc @ctomc

File: README.md [hunk 1]
```diff
@@ -12,12 +12,17 @@ This project provides the core runtime that is used by the Wildfly application s
 Building
 -------------------
 
-Ensure you have JDK 8 (or newer) and Maven 3.2.2 installed, then run:
+Ensure you have JDK 8 (or newer) and run:
```
Comment (ctomc) [inline README.md]: -Ensure you have JDK 8 (or newer) and optionally Maven 3.3.3 (or newer) installed, then run:

File: README.md [hunk 2]
```diff
@@ -12,12 +12,17 @@ This project provides the core runtime that is used by the Wildfly application s
...
 
-> mvn install
+> ./mvnw install
```
Comment (ctomc) [inline README.md]: mvn install if you use your own maven
or 
./mwn install if you dont

Comment (ctomc) [inline README.md]: also ./mwn wont work on windows

**Comment by ppalaga** [comment]
I force-pushed a new iteration ending with f7efb06 . I changed the README towards what was proposed by @ctomc : own mvn installation is presented as default, and Windows CLI syntax is covered. https://github.com/wildfly/wildfly-core/pull/2017/commits/71af845c599246e24df969c6dde2d0cc046e8eb4#diff-04c6e90faac2675aa89e2176d2eec7d8R15

Is this good to merge like this?

**Comment by rsvoboda** [comment]
WFCORE Jira is missing

**Comment by ppalaga** [comment]
> WFCORE Jira is missing

Yes, it is. As far as I can see, Jira-less commits are quite common in the history of commons.

**Comment by ctomc** [comment]
this is ok to test

**Comment by ppalaga** [comment]
In 493b4f8 , both commits are squashed into one, and the commit message was reworded to contain the newly created Jira https://issues.jboss.org/browse/WFCORE-2109

**Comment by jorsol** [comment]
There is an error in the  `mvnw.cmd` file with a comment:
https://github.com/takari/maven-wrapper/commit/d89e40a2f5dc67fc75603a213b945de2a5ac0b14

**Comment by ppalaga** [comment]
> There is an issue in the mvnw.cmd file with a comment: takari/maven-wrapper@d89e40a

@jorsol thanks for pointing this out. I fixed it manually in fc9367d 

cc https://github.com/takari/maven-wrapper/pull/27

**Pull request merged**